### PR TITLE
[Storage Blobs Service] Rename Results and Options

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**

### CODE REVIEW

File: sdk/storage/inc/blobs/blob_client.hpp [hunk 1]
```diff
@@ -241,24 +241,24 @@ namespace Azure { namespace Storage { namespace Blobs {
...
      * @param options Optional parameters to execute this function.
-     * @return A DownloadBlobToResult describing the downloaded blob.
+     * @return A DownloadToBlobResult describing the downloaded blob.
      */
-    Azure::Core::Response<DownloadBlobToResult> DownloadTo(
+    Azure::Core::Response<DownloadToBlobResult> DownloadTo(
         uint8_t* buffer,
         std::size_t bufferSize,
-        const DownloadBlobToOptions& options = DownloadBlobToOptions()) const;
+        const DownloadToBlobOptions& options = DownloadToBlobOptions()) const;
```
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_client.hpp:249]: DownloadToBlob is really misleading. I would rather have inconsistency than having this name.

Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_client.hpp:249]: Do you think `UploadFromBlockBlob` is misleading?

Comment (katmsft) [reply sdk/storage/inc/blobs/blob_client.hpp:249]: Yes, very. `UploadBlockBlobFrom` is way better.

Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_client.hpp:249]: This is proposed by Jeffrey in the Email thread. Why not bring it up if you're not okay with it?

Comment (katmsft) [reply sdk/storage/inc/blobs/blob_client.hpp:249]: Email sent to get Jeffrey's opinion on this.

Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/blob_client.hpp]: fixed

**Pull request merged**